### ISSUE TITLE
`throw undefined` turns into a `TypeError: Cannot read property 'message' of undefined`

### ISSUE THREAD

**Issue by mbest** [jsdom/jsdom#3183]
### Basic info:

- **Node.js version:** 12.16.1
- **jsdom version:** 16.5.1

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const dom = new JSDOM(`<div id="test"></div>`);
const test = dom.window.document.getElementById("test");
test.onclick = () => { throw undefined; };
test.click();
```

result:
```
Uncaught TypeError: Cannot read property 'message' of undefined
    at reportException (E:\Code\XS.AMC.UX\node_modules\jsdom\lib\jsdom\living\helpers\runtime-script-errors.js:58:84)
    at innerInvokeEventListeners (E:\Code\XS.AMC.UX\node_modules\jsdom\lib\jsdom\living\events\EventTarget-impl.js:341:9)
    at invokeEventListeners (E:\Code\XS.AMC.UX\node_modules\jsdom\lib\jsdom\living\events\EventTarget-impl.js:274:3)
    at HTMLDivElementImpl._dispatch (E:\Code\XS.AMC.UX\node_modules\jsdom\lib\jsdom\living\events\EventTarget-impl.js:221:9)
    at fireAnEvent (E:\Code\XS.AMC.UX\node_modules\jsdom\lib\jsdom\living\helpers\events.js:18:36)
    at HTMLDivElementImpl.click (E:\Code\XS.AMC.UX\node_modules\jsdom\lib\jsdom\living\nodes\HTMLElement-impl.js:79:5)
    at HTMLDivElement.click (E:\Code\XS.AMC.UX\node_modules\jsdom\lib\jsdom\living\generated\HTMLElement.js:110:34)
```

### How does similar code behave in browsers?

![image](https://user-images.githubusercontent.com/1230364/116932305-ee2b3880-ac16-11eb-809d-ec5df44d6d96.png)

### Why is this important?

We're trying to test a React component that throws `undefined` and is caught by an error boundary. But instead of the error boundary seeing `undefined`, it sees an `Error` with a message of `Cannot read property 'message' of undefined`.

**Comment by TimothyGu** [comment]
If this is an important use case for you, could you submit a pull request to fix the function in the backtrace to deal better with `undefined`?

https://github.com/jsdom/jsdom/blob/04f6c13f4a4d387c7fc979b8f62c6f68d8a0c639/lib/jsdom/living/helpers/runtime-script-errors.js#L38-L39

**Comment by mbest** [comment]
Thanks for the quick reply. I'm not familiar with the jsdom code, but I can take a stab at it.

**Comment by mbest** [comment]
Published in [16.6.0](https://github.com/jsdom/jsdom/blob/16.6.0/Changelog.md)
